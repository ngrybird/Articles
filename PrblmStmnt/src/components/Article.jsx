import React from "react";

const Article = props => (
  <>
    <tr>
      <td className="article-item">{props.articleData.title}</td>

      <td className="article-item">{props.articleData.upvotes}</td>
      <td className="article-item">
        {new Date(props.articleData.date).toISOString()}
      </td>
    </tr>
  </>
);

export default Article;
