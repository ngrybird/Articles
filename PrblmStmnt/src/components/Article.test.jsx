import React from "react";
import Article from "./Article";
import { fireEvent, render } from "@testing-library/react";
import { articles } from "../utils/Data";

describe("<Article/>", () => {
  it("render Article component with sample data", () => {
    const { container, getAllByRole } = render(
      <Article articleData={articles[0]} />
    );
    expect(container).toMatchSnapshot();
  });
});
