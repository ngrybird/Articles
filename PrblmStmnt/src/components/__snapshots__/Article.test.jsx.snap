// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Article/> render Article component with sample data 1`] = `
<div>
  <tr>
    <td
      class="article-item"
    >
      Harry Potter and the Sorcererâ€™s Stone Review
    </td>
    <td
      class="article-item"
    >
      56
    </td>
    <td
      class="article-item"
    >
      2016-12-22T18:30:00.000Z
    </td>
  </tr>
</div>
`;
